spearhead/assets/sass/block-extends/_search.scss

```diff
@@ -5,53 +5,64 @@
 
 .wp-block-search {
 
-	// wrong class on gutenberg editor, we'll need to change it after that's fixed
-	&.wp-block-search__icon-button.wp-block-search__button-inside {
-		.wp-block-search__button {
-			padding: var(--button--padding-vertical) var(--button--padding-horizontal);
-		}
-	}
-
-	// wrong class on gutenberg editor, we'll need to change it after that's fixed
-	&.wp-block-search__text-button.wp-block-search__button-inside {
+	flex-wrap: wrap;
 
+	&.wp-block-search__button-inside {
 		.wp-block-search__inside-wrapper{
-			flex-direction: row-reverse;
-			padding-left: 0;
-			border: none;
-			border-bottom: var(--form--border-width) solid var(--form--border-color);
+			padding: 0;
+			.wp-block-search__button {
+				padding: var(--button--padding-vertical) var(--button--padding-horizontal);
+				border-top-left-radius: 0;
+				border-bottom-left-radius: 0;
+				margin: -1px -1px -1px 0;
+			}
 		}
 
-		.wp-block-search__input {
-			border: none;
-			padding: 0 0 0 10px;
-			font-size: var(--global--font-size-sm);
+		&:not(.wp-block-search__icon-button) {
+			.wp-block-search__inside-wrapper{
+				border-radius: var(--button--border-radius);
+			}
 		}
 
-		.wp-block-search__button {
-			&.has-icon {
-				margin-left: 0;
-				background: transparent;
-				color: var(--entry-header--color);
-				padding: 0;
+		&.wp-block-search__icon-button {
+
+			.wp-block-search__inside-wrapper{
+				flex-direction: row-reverse;
```

Comment:
> These look pretty good! I'm seeing just a couple bugs with the "Button inside" configuration, both with and without the search icon:

@kjellr  the buttons should always be on the right side (my screenshots are on my local install with the gutenberg plugin downloaded from the repo directly installed), this bit of the code is what moves them to the right. Maybe you don't have the latest version? The old Gutenberg version didn't have the `.wp-block-search__icon-button` class correctly showing up

Reply:
Ok, it's mostly working now. Using the `master` for Gutenberg, the only bug I'm seeing is that I'm unable to get the Button inside with search icon configuration working: 

![search-button](https://user-images.githubusercontent.com/1202812/96169748-9fbf8200-0ef0-11eb-85dc-97706dcb1246.gif)

Is that due to the bug you're mentioning?

Reply:
No, that's the intended style for button inside, icon instead of text. If you are looking for this, then this is button **outside**, button with an icon (and I believe that you have the correct gutenberg version now):

<img width="757" alt="Screenshot 2020-10-15 at 20 22 58" src="https://user-images.githubusercontent.com/3593343/96170794-59841600-0f24-11eb-99af-6aff34867733.png">

Reply:
Ha, you're right! Thanks for explaining. It's a little confusing, but it makes sense now.